pkg/kubectl/resource/builder.go

```diff
@@ -402,6 +402,7 @@ func (b *Builder) Do() *Result {
 	if b.requireNamespace {
 		helpers = append(helpers, RequireNamespace(b.namespace))
 	}
+	helpers = append(helpers, FilterNamespace())
```

Comment:
Add a todo that kubectl should probably warn / prevent the user from creating different type of resources at the same time.